src/Demo.tsx

```diff
@@ -0,0 +1,173 @@
+import * as React from "react";
+import Panel from "./components/Panel";
+import Header from "./components/Header";
+import Form from "./components/Form";
+import Fieldset from "./components/Fieldset";
+import Input from "./components/Input";
+import Tabs from "./components/Tabs";
+import SubmitButton from "./components/SubmitButton";
+
+export interface DemoState {
+  lights: boolean;
+  errorText: string;
+  successText: string;
+  infoText: string;
+}
+
+export default class Demo extends React.Component<void, DemoState> {
+  constructor(props) {
+    super(props);
+    this.state = { lights: true, errorText: null, successText: null, infoText: null };
+    this.toggleLights = this.toggleLights.bind(this);
+    this.makeTabItems = this.makeTabItems.bind(this);
+    this.setError = this.setError.bind(this);
+    this.setSuccess = this.setSuccess.bind(this);
+    this.setInfo = this.setInfo.bind(this);
+  }
+
+  makeTabItems() {
+    const COLORS = {
+      "Red": {
+        "#D0343A": "$red",
+        "#97272C": "$red_dark",
+        "#DE7175": "$red_tint",
+        "#F6D6D8": "$red_error"
+      },
+      "Orange/Yellow": {
+        "#FFB81D": "$orange",
+        "#D2A74A": "$orange_desaturated",
+        "#FEE24A": "$yellow",
+        "#FFF1A5": "$yellow_tint"
+      },
+      "Green": {
+        "#799A05": "$green",
+        "#497629": "$green_dark",
+        "#008918": "$green_bright",
+        "#F3F7E6": "$green_tint"
+      },
+      "Blue": {
+        "#07818D": "$teal",
+        "#047074": "$teal-dark",
+        "#83C0C6": "$teal-tint",
+        "#1B7FA7": "$blue",
+        "#135772": "$blue-dark",
+        "#73C7E8": "$light-blue",
+        "#D1E5ED": "$blue-tint"
+      },
+      "Gray": {
+        "#776E64": "$gray",
+        "#D7D4D0": "$light-gray",
+        "#DDDDDD": "$medium-gray",
+        "#080807": "$dark-gray",
+        "#54514A": "$gray-brown",
+        "#111111": "$black",
+        "#F5F4F3": "$gray-tint"
+      }
+    };
+
+    let items = {};
+    Object.keys(COLORS).map((colorCategory) => {
```

Comment:
`map` returns an array which you are not assigning a variable to so a for loop is better.